components/credentials/src/main/java/org/cloudfoundry/credhub/generators/CertificateGenerator.java

```diff
@@ -23,17 +24,23 @@
   private final RsaKeyPairGenerator keyGenerator;
   private final SignedCertificateGenerator signedCertificateGenerator;
   private final CertificateAuthorityService certificateAuthorityService;
+  private final Integer caMinimumDuration;
+  private final Integer leafMinimumDuration;
 
 
   @Autowired
   public CertificateGenerator(
     final RsaKeyPairGenerator keyGenerator,
     final SignedCertificateGenerator signedCertificateGenerator,
-    final CertificateAuthorityService certificateAuthorityService) {
+    final CertificateAuthorityService certificateAuthorityService,
+    @Value("${certificates.ca_minimum_duration:#{null}}") final Integer caMinimumDuration,
+    @Value("${certificates.leaf_minimum_duration:#{null}}") final Integer leafMinimumDuration) {
     super();
     this.keyGenerator = keyGenerator;
     this.signedCertificateGenerator = signedCertificateGenerator;
     this.certificateAuthorityService = certificateAuthorityService;
+    this.caMinimumDuration = caMinimumDuration;
+    this.leafMinimumDuration = leafMinimumDuration;
```

Comment:
And on these lines, the values would be converted form the configuration Integer to the backend's Duration.